usm_rest_api/v1/views/views.py

```diff
@@ -0,0 +1,371 @@
+The resource is used to create a cluster with a list of hosts by posting a
+message as follows:
+
+    {
+    "cluster_name": "NAME",
+    "cluster_type": "TYPE",
+    "storage_type": "FSTYPE",
+    "nodes": [
+        {
+            "node_name": "NAME",
+            "management_ip": "ADDR",
+            "cluster_ip": "ADDR",
+            "public_ip": "ADDR",
+            "ssh_username": "USER",
+            "ssh_password": "PASS",
+            "ssh_key_fingerprint": "FINGER",
+            "ssh_port": 22,
+            "node_type": "nodetype"
+        },
+        {
+            
+        }
+    ]
+}
+    """
+    log.debug("Inside create_cluster. Request Data: %s" % request.data )
+    data = request.data.copy()
+    #
+    #TODO:Check the type of the cluster to be created and call appropriate task
+    #
+    #create gluster cluster
+    jobId = createGlusterCluster.delay(data)
+    log.debug("Exiting create_cluster JobID: %s" % jobId)
+    return Response(str(jobId), status=200) 
+
+
+@shared_task
+def createGlusterCluster(postdata):
+    log.debug("Inside createGlusterCluster Async Task")
+    failedNodes=[]
+    nodelist = postdata['nodes']
+    noOfNodes = len(nodelist)
+    del postdata['nodes']
+    
+     #create the cluster
+    createGlusterCluster.update_state(state='CREATE_CLUSTER')
+    postdata['cluster_id'] = uuid.uuid4()
+    try:
+        clusterSerilaizer = ClusterSerializer(data=postdata)
+        if clusterSerilaizer.is_valid():
+            clusterSerilaizer.save()
+        else:
+                log.error("Cluster Creation failed. Invalid clusterSerilaizer")
+                log.error("clusterSerilaizer Err: %s" % clusterSerilaizer.errors)
+                return {'status':'CLUSTER_CREATION_FAILURE','error':clusterSerilaizer.errors}
+    except Exception, e:
+        log.exception(e)
+        return {'status':'CLUSTER_CREATION_FAILURE','error':str(e)}
+   
+    #Create the Nodes in the cluster
+    #Setup the host communication channel
+    createGlusterCluster.update_state(state='ESTABLISH_HOST_COMMUNICATION')
+    for node in nodelist[:]:
+        try:
+            ssh_fingerprint = saltapi.get_fingerprint(
+                saltapi.get_host_ssh_key(node['management_ip']))
+            rc,out,err = saltapi.setup_minion(node['management_ip'],
+                                              ssh_fingerprint, node['ssh_username'],
+                                              node['ssh_password'], postdata)
+            if rc!=0:
+                raise Exception("Accept Minion Failed", rc,out,err)
+        except Exception, e:
+            log.exception(e)
+            #Delete the node from list to avoid further processing
+            nodelist.remove(node)
+            failedNodes.append(node)
+    #Sleep for sometime to make sure all the restarted minions are back
+    #online
+    log.debug("Accepting the minions keys" )
+    createGlusterCluster.update_state(state='ADD_MINION_KEYS_AND_DB_UPDATE')
+    time.sleep( 3 )
```

Comment:
This is a temporary arrangement to wait till minion is Up and running. Now Bala has re-factored the Back-end code such that this wait is not required. I will update this once h pushes his code into the master